internal/email/email.go

```diff
@@ -61,56 +108,84 @@ func NewMessage(
 	return message.NewMessage(id.String(), payload), nil
 }
 
-// getBodyHTML returns the HTML body for the email based on the message payload
-func getEmailBodyHTML(ctx context.Context, inviteURL, minderURL, sponsor, project, role, inviteeEmail string) string {
-	data := struct {
-		AdminName        string
-		OrganizationName string
-		InvitationURL    string
-		RecipientEmail   string
-		MinderURL        string
-		TermsURL         string
-		PrivacyURL       string
-		SignInURL        string
-		RoleName         string
-		RoleVerb         string
-	}{
-		AdminName:        sponsor,
-		OrganizationName: project,
-		InvitationURL:    inviteURL,
-		RecipientEmail:   inviteeEmail,
-		MinderURL:        minderURL,
-		TermsURL:         DefaultMinderTermsURL,
-		PrivacyURL:       DefaultMinderPrivacyURL,
-		SignInURL:        minderURL,
-		RoleName:         role,
-		RoleVerb:         authz.AllRolesVerbs[authz.Role(role)],
-	}
-
-	// TODO: Load the email template from elsewhere
+// getHTMLBodyString returns the HTML body for the email based on the message payload.
+// If there is an error creating the HTML body, it will try to return the text body instead
+func getEmailBodyHTML(ctx context.Context, data bodyData) (string, error) {
+	var b strings.Builder
+	bHTML := bodyHTML
 
-	// Parse the template
-	tmpl, err := templates.Parse(bodyHTML)
+	bodyHTMLTmpl, err := util.NewSafeHTMLTemplate(&bHTML, "body-invite-html")
 	if err != nil {
-		zerolog.Ctx(ctx).Error().Err(err).Msg("error parsing the HTML template for email invitations")
-		// Default to the text body
-		return getEmailBodyText(inviteURL, sponsor, project, role)
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error creating the HTML template for email invitations")
+		return getEmailBodyText(ctx, data)
 	}
-	// Execute the template
-	var b strings.Builder
-	if err := tmpl.Execute(&b, data); err != nil {
-		return ""
+	err = bodyHTMLTmpl.Execute(ctx, &b, data, EmailBodyMaxLength)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error executing the HTML template for email invitations")
+		return getEmailBodyText(ctx, data)
```

Comment:
Ditto.

Note that if you switch this to a `return "", err`, then you'll be returning an error, and won't need the log line above.  (Currently, you're _handling_ the error by attempting to call the text template.)